modules/java/generator/gen_java.py

```diff
@@ -1535,7 +1538,7 @@ def isSmartClass(self, classname):
         '''
         Check if class stores Ptr<T>* instead of T* in nativeObj field
         '''
-        return self.isWrapped(classname) and self.classes[classname].base
+        return False
```

Comment:
Sorry I was on vacations. This method was causing that objects are passing to java as Ptr<T>  whereas in java it was casted to T directly, which was not working at all. 

BTW. It seems that there is a memory leak issue there so some further work is needed there. 

https://github.com/opencv/opencv/pull/6815/files/ef450050561b9107f3cb7eff57e27917dc4165dc#r76603254

I will try to think about it and change it later this week.